code/modules/interface/keybind_menu_chui.dm

```diff
@@ -1,51 +1,52 @@
 
-chui/window/keybind_menu
-	name = "Keybinding Customization"
+/datum/keybind_menu
 	var/client/owner
-	windowSize = "700x550"
 	var/list/changed_keys //so we can keep track of what keys the user changes then merge later
-	var/datum/keymap/current_keymap
-	var/last_interact_time //To rate-limit
-	var/const/rate_limit_time = 1 SECOND
+	var/resetting = FALSE
+	var/hasChanges = FALSE
 
 	New(client/my_client)
 		..()
 		owner = my_client
-		theAtom = owner.mob
-		current_keymap = owner.keymap
 
-	Subscribe(client/who)
-		..()
-		changed_keys = list() //new one every time so we're not merging an ever-expanding list (feel free to change if cheaper)
-
-	GetBody()
-
-		current_keymap = owner.keymap //Need to refresh this since it's not a proper pointer?
-
-		var/list/html = list()
+	ui_interact(mob/user, datum/tgui/ui)
+		ui = tgui_process.try_update_ui(user, src, ui)
+		if (!ui)
+			ui = new(user, src, "Keybinds", "Keybinding Customization")
```

Comment:
Stick the title on the actual UI instead, I guess. It's how we do the vast majority, and this one doesn't have a special reason to customize the title or whatever.